### PR TITLE
[KYUUBI #1581] Add SchemaHelper for trino engine

### PR DESCRIPTION
**Pull request by hddong**
### _Why are the changes needed?_

Add SchemaHelper for trino engine

### _How was this patch tested?_
- [X] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [X] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/schema/SchemaHelper.scala [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    case VARCHAR => TTypeId.VARCHAR_TYPE
+    case VARBINARY => TTypeId.BINARY_TYPE
+    case DATE => TTypeId.DATE_TYPE
+    case TIMESTAMP => TTypeId.TIMESTAMP_TYPE
+    case INTERVAL_DAY_TO_SECOND => TTypeId.INTERVAL_DAY_TIME_TYPE
+    case INTERVAL_YEAR_TO_MONTH => TTypeId.INTERVAL_YEAR_MONTH_TYPE
+    case ARRAY => TTypeId.ARRAY_TYPE
+    case MAP => TTypeId.MAP_TYPE
+    case ROW => TTypeId.STRUCT_TYPE
+    case _ => TTypeId.STRING_TYPE
```
Comment (yaooqinn) [inline externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/schema/SchemaHelper.scala:58]: shall we raise an error？

Comment (hddong) [reply externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/schema/SchemaHelper.scala]: Done.

File: externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/schema/SchemaHelper.scala [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+import org.apache.hive.service.rpc.thrift.TTableSchema
+import org.apache.hive.service.rpc.thrift.TTypeDesc
+import org.apache.hive.service.rpc.thrift.TTypeEntry
+import org.apache.hive.service.rpc.thrift.TTypeId
+import org.apache.hive.service.rpc.thrift.TTypeQualifiers
+import org.apache.hive.service.rpc.thrift.TTypeQualifierValue
+
+object SchemaHelper {
+
+  private lazy val STRING_TYPES = Array(
```
Comment (yaooqinn) [inline externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/schema/SchemaHelper.scala:39]: let's use set?

File: externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/schema/SchemaHelper.scala [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    case DATE => TTypeId.DATE_TYPE
+    case TIMESTAMP => TTypeId.TIMESTAMP_TYPE
+    case INTERVAL_DAY_TO_SECOND => TTypeId.INTERVAL_DAY_TIME_TYPE
+    case INTERVAL_YEAR_TO_MONTH => TTypeId.INTERVAL_YEAR_MONTH_TYPE
+    case ARRAY => TTypeId.ARRAY_TYPE
+    case MAP => TTypeId.MAP_TYPE
+    case ROW => TTypeId.STRUCT_TYPE
+    case stringType if STRING_TYPES.contains(stringType) => TTypeId.STRING_TYPE
+    case other =>
+      throw new IllegalArgumentException(s"Unrecognized type name: $other")
```
Comment (yaooqinn) [inline externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/schema/SchemaHelper.scala:74]: Unrecognized presto/trino data type: ?

Comment (hddong) [reply externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/schema/SchemaHelper.scala]: Thanks, have addressed them.

**Pull request closed without merging**